Merge в master от 12.04.2023

ZhosanDenis opened the pull request.

Garik-create committed:
Добавлены эндпойнты по поиску отчётов по айди собаки и по айди собаки и дате отчёта.

Dmitriy1604 committed:
коммит

Dmitriy1604 committed:
создал таймер который отправляет сообщения о том что отчет просрочен, создал поле chatId у carer и в scripts.sql

KlochkoVitaliy committed:
Дополнение API на контроллеры

Garik-create committed:
Добавлен поиск опекуна по номеру телефона. Скорректированы связи между таблицами oneToOne - OneToMany.

ZhosanDenis committed:
Изменения №1 от 11.04.2023
- в классе Agreement изменен тип поля number (int -> String). Также изменен тип данных колонки number таблицы БД agreements;
- в application.properties добавлен параметр с путем сохранения фотографий, отправленных боту в качестве отчета;
- в классе Carer добавлены геттеры и сеттеры для полей dog, agreement, dailyReport;
- в классе CarerController в эндпоинтах изменен тип возвращаемых объектов (Carer -> CarerRecord);
- в классе CarerRecord конструктор с аргументами изменен на конструктор по умолчанию;
- в CarerRepository добавлен метод по поиску опекуна по номеру договора;
- в CarerService добавлена инжекция маппера, использованы методы маппера для конвертации Carer в CarerRecord и обратно;
- в DailyReport добавлены геттеры и сеттеры для полей carer, dog;
- в Dog добавлены геттеры и сеттеры для полей carer, dailyReport;
- в классе DogController в эндпоинтах изменен тип возвращаемых объектов (Dog -> DogRecord);
- в классе DogRecord конструктор с аргументами изменен на конструктор по умолчанию;
- в DogService добавлена инжекция маппера, использованы методы маппера для конвертации Carer в CarerRecord и обратно;
- создан класс ModelMapper для конвертации сущностей в Record и обратно;
- в scripts.sql добавлена таблица volunteer_chats. Созданы VolunteerChat и VolunteerChatRepository для хранения id чата волонтеров -> при изменения чата значение перезаписывается;
- в TelegramBotUpdatesListener добавлена новая команда для отправки отчета о питомце. Добавлено поле для хранения пути, куда сохраняются фото питомца. Добавлена инжекция VolunteerChatRepository. Обработки команд (начинающихся с /) и обычных текстовых сообщений вынесены в отдельные методы. Добавлен метод по сохранению фото, присланных через бота (пока только на компьютере). Реализована обработка приема отчета о питомце. Изменен принцип хранения id чата волонтеров - теперь хранится не в поле, а в переменной, которая постоянно делает запрос в БД (в случае изменения чата, id чата перезаписывается).

ZhosanDenis committed:
Изменения №2 от 11.04.2023
- в TelegramBotUpdatesListener вернул поле VOLUNTEER_CHAT_ID для устранения конфликта при merge в dev-ветку.

Garik-create committed:
Сменил logger на LOGGER

ZhosanDenis committed:
Изменения №1 от 12.04.2023
- в TelegramBotUpdatesListener добавлена дополнительная проверка на случай, когда опекун прислал документ. Изменено формирование имени чата волонтеров - теперь добавляется дата создания полностью, а не только месяц и год

KlochkoVitaliy committed:
Дополнена информация по JavaDoc

KlochkoVitaliy committed:
Дополнена информация по JavaDoc + дополнил @throws

ZhosanDenis committed:
Изменения №2 от 12.04.2023
- добавлен enum-класс со всеми командами для телеграм бота. Данные перечисления заменяют поля-константы TelegramBotUpdatesListener;
- в ModelMapper добавлено приведение ФИО опекуна к нижнему регистру и написание частей ФИО с заглавной буквы;
- в pom.xml добавлена зависимость для библиотеки org.apache.commons.lang3, чтобы использовать StringUtils;
- в TelegramBotConfiguration в SetMyCommands() заменено использование команд-констант из TelegramBotUpdatesListener на новые BotCommand с данными из enum-класса Command;
- в TelegramBotUpdatesListener добавлена отправка кнопки в чат волонтеров, которая отправляется вместе с сообщением о необходимости связаться с клиентом и ссылкой на него. При подтверждении в чат с волонтерами отправляется сообщение с благодарностью - тем самым можно увидеть, связывался ли кто-то с клиентом или нет.

ZhosanDenis committed:
Изменения №3 от 12.04.2023
- удален комментарий о необходимости убрать все поля-константы в TelegramBotUpdatesListener для предотвращения конфликта merge с веткой dev.

ZhosanDenis committed:
Merge pull request #8 from Zhoss/1-feature-Dima

1 feature dima

KlochkoVitaliy committed:
Внесены изменения в Проперти

Garik-create committed:
Merge branch 'dev' into 1-feature-igor

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java

Garik-create committed:
Merge pull request #11 from Zhoss/1-feature-igor

1 feature igor

KlochkoVitaliy committed:
Merge branch 'dev' into 2-feature-Vitaliy

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java

ZhosanDenis committed:
Merge pull request #10 from Zhoss/2-feature-Vitaliy

2 feature vitaliy

ZhosanDenis committed:
Merge branch 'dev' into 1-feature-denis

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DogController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/model/Carer.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/repository/CarerRepository.java
#	src/main/resources/application.properties
#	src/main/resources/liquibase/scripts/scripts.sql

ZhosanDenis committed:
Merge pull request #9 from Zhoss/1-feature-denis

Изменения №1 от 11.04.2023

Garik-create committed:
Скорректировал код. Код рабочий.

Garik-create committed:
Merge pull request #13 from Zhoss/1-feature-igor

Скорректировал код. Код рабочий.

KlochkoVitaliy committed:
Дополнены API и JavaDoc

Garik-create committed:
Сделал стандартные эндпойнты к договорам.

ZhosanDenis committed:
Merge pull request #14 from Zhoss/1-feature-igor

Сделал стандартные эндпойнты к договорам.

ZhosanDenis committed:
Изменения №1 от 14.04.2023
- класс Agreement: изменена аннотация @OneToOne, добавлены проверки для setNumber() и setConclusionDate();
- класс Carer: добавлено поле passportNumber + геттер/сеттер, удалено поле agreement, добавлены проверки для setPhoneNumber() и setPassportNumber();
- CarerController: удален эндпоинт по добавлению опекуна, POST-запрос editCarer() заменен на PATCH-запрос;
- класс CarerRecord: добавлены поля passportNumber и dogId, а также геттеры и сеттеры для них;
- CarerRepository: метод findCarerByAgreementNumber() заменен на findCarerByChatId();
- CarerService: удален метод addCarer(), метод findCarer() заменен на findCarerByChatId();
- класс DailyReport: удалены поля description, photo и dog;
- DailyReportController: удалены эндпоинты для поиска отчетов по id собаки, добавлен эндпоинт для загрузки фото из отчета (по id опекуна и дате);
- DailyReportRepository: удалены методы findDailyReportsByDogId() и findDailyReportsByDogIdAndReportDateIs(), метод findDailyReportByCarerIdAndReportDate() возвращает один отчет, а не list;
- DailyReportService: удалены методы findDailyReportsByDogId() и findDailyReportByDogAndReportDate(), добавлен метод addDailyReport();
- класс Dog: удалены поля carer и dailyReport;
- таблицы БД: удалены и добавлены колонки согласно изменениям в сущностях выше;
- TelegramBotUpdatesListener: удалены неиспользуемые поля-константы с чатом волонтеров и командами (которые переместились в Enum-класс Command), добавлено сохранение в БД параметров фото для отчета.

ZhosanDenis committed:
Merge branch 'dev' into 1-feature-denis

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/model/Agreement.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/ModelMapper.java

ZhosanDenis committed:
Изменения №2 от 14.04.2023
- AgreementController: добавлен эндпоинт для получения всех договоров, добавлен метод для обработки исключения с @ExceptionHandler
- AgreementRecord: добавлено поле id + геттер и сеттер;
- добавлен функционал изменения состоянии сущности Dog (поля isTaken и onProbation становятся true) при создании договора, добавлены методы для получения всех договоров и всех договоров за дату;
- в application.properties удалены параметры, хранящие id чатов клиента и волонтеров, т.к. данные id хранятся в БД;
- CarerController: добавлен эндпоинт по получению всех опекунов;
- CarerRepository: изменен тип возвращаемых данных методов findCarerByPhoneNumber() и findCarerByChatId() c Carer на Optional<Carer>;
- CarerService: добавлен метод для добавления опекуна через TelegramBotUpdatesListener, добавлен метод по получению всех опекунов;
- добавлен класс DailyReportRecord;
- DogController: добавлен эндпоинт по получению всех собак;
- DogService: добавлен метод по получению всех собак;
- TelegramBotUpdatesListener: убрана проверка присланного сообщения на соответствие паттерну номера договора (до этого опекун искался по номеру договора, сейчас по chatId), добавлены сохранения присылаемой информации по рациону, самочувствию и изменении поведения собаки в таблицу БД daily_reports, добавлен метод по созданию задач TimerTask и созданию напоминанию при добавлении договора(уточнить у наставника о работоспособности кода - не знаю как протестировать);
- Timer: закомментирован код, метод в TelegramBotUpdatesListener больше соответствует ТЗ.

ZhosanDenis committed:
Merge pull request #15 from Zhoss/1-feature-denis

Изменения №1 от 14.04.2023

BizinMitya commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -123,115 +110,270 @@ public int process(List<Update> updates) {
+                    LocalDate.now() + "." + mediaType);
+            try {
+                URL url = new URL(fullPath);
+                Files.createDirectories(filePath.getParent());
```
не нужно руками скачивать, у библиотеки есть метод для скачивания

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -123,115 +110,270 @@ public int process(List<Update> updates) {
+            Path filePath = Path.of(photosDir + "/" + carer.getFullName(),
+                    LocalDate.now() + "." + mediaType);
+            try {
+                URL url = new URL(fullPath);
```
не нужно руками скачивать, у библиотеки есть метод для скачивания

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/model/VolunteerChat.java:
```diff
@@ -0,0 +1,47 @@
+public class VolunteerChat {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    long id;
```
Все поля должны быть приватные

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -0,0 +1,74 @@
+        agreementRepository.deleteById(id);
+    }
+
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -0,0 +1,74 @@
+        return modelMapper.mapToAgreementRecord(agreement);
+    }
+
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -62,34 +54,68 @@ public Carer addCarer(String fullName, int age, String phoneNumber) {
     }
 
     @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -62,34 +54,68 @@ public Carer addCarer(String fullName, int age, String phoneNumber) {
+     * @throws CarerNotFoundException если опекун с таким идентификационным номером (id) не найден
+     * @see org.springframework.data.jpa.repository.JpaRepository#findById(Object)
+     */
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -62,34 +54,68 @@ public Carer addCarer(String fullName, int age, String phoneNumber) {
+     * @throws IllegalArgumentException Если поля <b>carerRecord</b> пустые (null)
+     * @see CarerRecord
+     */
     @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -101,7 +127,24 @@ public void deleteCarer(long id) {
 
-    public boolean existsCarerByFullNameAndPhoneNumber(String fullName, String phoneNumber) {
-        return this.carerRepository.existsCarerByFullNameAndPhoneNumber(fullName, phoneNumber);
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -101,7 +127,24 @@ public void deleteCarer(long id) {
+        }
+    }
+
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/DogService.java:
```diff
@@ -9,60 +9,89 @@
+     * @throws DogNotFoundException Если нет информации в БД.
+     * @see org.springframework.data.jpa.repository.JpaRepository
+     */
     @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/DogService.java:
```diff
@@ -72,4 +101,11 @@ public void deleteDog(long id) {
         }
     }
+
+    @Transactional
```
readOnly=true лучше поставить, т.к. только чтение

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -395,4 +545,62 @@ public void sendCallVolunteerCommand(long chatId,
+    public void sendNotification() {
+        List<Agreement> agreements = this.agreementService.findAllAgreementsByDate(LocalDate.now());
+        agreements.forEach(agreement -> {
+            this.timer = new Timer();
```
здесь не нужен никакой таймер, зачем он? у тебя аннотация @Scheduled стоит, которая уже говорит о том, что метод надо вызывать периодически. Также нужна аннотация @EnableScheduling, над главным классом её можно повесить.

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -395,4 +545,62 @@ public void sendCallVolunteerCommand(long chatId,
+            this.timer = new Timer();
+            this.timer.scheduleAtFixedRate(new TimerTask() {
+                @Autowired
+                private VolunteerChatRepository volunteerChatRepository;
```
Это надо инжектить в поле класса TelegramBotUpdateListener и просто использовать внутри метода

KlochkoVitaliy committed:
Добавлены тесты на DogController внесены дополнения в DogService

KlochkoVitaliy committed:
Исправлены тесты на DogController

KlochkoVitaliy committed:
Merge branch '2-feature-Vitaliy' into dev

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/repository/CarerRepository.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/DailyReportService.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/ModelMapper.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/timer/Timer.java

ZhosanDenis committed:
Изменения №1 от 17.04.2023
- добавлены tags и подкорректированы описания эндпоинтов AgreementController, CarerController, DailyReportController, DogController;
- добавлены недостающие логирование и проверки в AgreementService, CarerService, DailyReportService, DogService;
- добавлены модификаторы доступы private полям класса VolunteerChat.

ZhosanDenis committed:
Merge branch 'dev' into 1-feature-denis

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java

ZhosanDenis committed:
Merge pull request #17 from Zhoss/1-feature-denis

Изменения №1 от 17.04.2023

Zhoss commented:
Merge approved
